beamer/agent/chain.py

```diff
@@ -476,6 +482,147 @@ def maybe_invalidate(claim: Claim, context: Context) -> None:
     claim.start_challenge()
 
 
+def _proof_ready_for_l1_relay(request: Request) -> bool:
+    return (
+        not request.l1_resolved.is_active
+        and request.fill_tx is not None
+        and request.fill_timestamp is not None
+    )
+
+
+def _invalidation_ready_for_l1_relay(claim: Claim) -> bool:
+    return (
+        not claim.invalidated_l1_resolved.is_active
+        and claim.invalidation_tx is not None
+        and claim.invalidation_timestamp is not None
+    )
+
+
+def _l1_resolution_threshold_reached(claim: Claim, context: Context) -> bool:
+    l1_gas_cost = 1_000_000  # TODO: Adapt to real price
```

Comment:
How far is the actual gas cost on L1 from this value?

Reply:
To extend on this a bit, costs actually depend on the two chains, which are part of the resolution. My analysis shows that `from_chain` (the contract where we get the message from), mainly determines the costs. 

Message flow example: 

Opt -> Arb 950k gas
ETH -> Arb 150k gas 

The reason for it is that getting the message from the Optimism outbox, you have to verify a merkle proof but on the EthereumL2Messenger, you just have to encode and hash some values. 

For now I don't see this as a problem as long as we choose a value higher than the highest costs, which is true.

Reply:
In any case we have to rework this calculation as we also should incorporate costs for the challenges being done, and we do this check on two different places in the code base (here and [here](https://github.com/beamer-bridge/beamer/blob/d4bbfaf966e0364df45213a66c865afb1e99407d/beamer/agent/chain.py#L441))

Reply:
yes in the other PR i guess i changed to use only one calculation